invenio_files_rest/models.py

```diff
@@ -1187,6 +1264,118 @@ def relink_all(cls, old_file, new_file):
             ObjectVersion.query.filter_by(file_id=str(old_file.id)).update({
                 ObjectVersion.file_id: str(new_file.id)})
 
+    def get_tags(self):
+        """Get tags for object version as dictionary."""
+        return {t.key: t.value for t in self.tags}
+
+    def __eq__(self, other):
+        """Check if the two object are equals."""
+        return other and isinstance(other, self.__class__) and \
+            self.key == other.key and self.file_id == other.file_id
+
+    def __ne__(self, other):
+        """Check if are not equal."""
+        return not self.__eq__(other=other)
+
+    def merge(self, bucket, key=None):
```

Comment:
I'm not sure I fully understand the use case for the merging support. It seems to add quite a lot of complexity overall. To discuss IRL.

Reply:
In videos, the merge method is called when re-publishing an already published deposit after edition. I guess this happens for a simple use case: we have transcoding videos files happening even if the record is published because it is a long process and we want to allow the generated video files to be added to the bucket after publishing.
So when you edit, we want to take the deposit bucket and merge it with the published one that might have new files.

Unfortunately, it is also causing issues. To be discussed yes.